src/main/Main.hs

```diff
@@ -498,18 +498,32 @@ interpreterHandler
   -> ParserFailure ParserHelp
   -> IO (GlobalOptsMonoid, (GlobalOpts -> IO (), t))
 interpreterHandler args f = do
-  isFile <- D.doesFileExist file
-  if isFile
-  then runInterpreterCommand file
-  else parseResultHandler (errorCombine (noSuchFile file))
+  -- args can include top-level config such as --extra-lib-dirs=... (set by
+  -- nix-shell) - we need to find the first argument which is a file, everything
+  -- afterwards is an argument to the script, everything before is an argument
+  -- to Stack
+  (stackArgs, fileArgs) <- spanM (fmap not . D.doesFileExist) args
```

Comment:
We can use `takeWhileM` from [monad-loops](https://hackage.haskell.org/package/monad-loops-0.4.3/docs/Control-Monad-Loops.html). `monad-loops` is already a dependency of stack via `monad-logger`.

The `spanM` function can be added to `monad-loops` package.